apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -275,13 +275,7 @@ namespace MWRender
         auto lightingMethod = SceneUtil::LightManager::getLightingMethodFromString(Settings::Manager::getString("lighting method", "Shaders"));
 
         resourceSystem->getSceneManager()->setParticleSystemMask(MWRender::Mask_ParticleSystem);
-        // Shadows and radial fog have problems with fixed-function mode
```

Comment:
Move the comment as well as the code please :)

Reply:
The code below tells you the same if not more. There are two more features not working with fixed function mode that the comment failed to mention.

Reply:
The existing comment isn't great, but we've had confusion with this before when people first encounter a setting called `force shaders` controlling a variable called `force shaders` that's also controlled by other things. A new, better, comment could help, but if I write it, it's likely to be confusingly verbose.

The gist is that there are some settings that mean we might require shaders for specific nodes, like `auto use normal maps`, and some settings that require shaders for everything, like `enable shadows`. We've also got users on Intel GPUs that aren't remotely close to spec-compliant for the FFP who need shaders for everything, people using modded lighting shaders, and developers who want to compare shaders and the FFP with no bells and whistles throwing off the comparison.

The setting `force shaders` is meant to mean *force shaders to be used everywhere, even if the FFP can theoretically handle a given draw call*, and is meant for the buggy drivers, vtastek-users, and developers testing things. When it's `false`, the default, we'll only use shaders for draw calls that need shaders.

The variable by the same name is meant to mean *we need to use shaders for everything for whatever reason* and doesn't make any distinction between it being specifically requested by the user or implied by another setting being on.

We've also had people wanting a `force fixed function` setting that overrides *all* the settings that might make shaders get used, as that would be easier for testing than remembering which settings might use shaders and disabling all of them.

Reply:
Probably we should address these confusions in the settings documentation. From the code perspective what we do seems obvious.

Reply:
The documentation for `force shaders` itself doesn't actually mention that enabling certain other settings will implicitly turn it on, and currently just mentions bump maps, so it could stand to be improved if you're up for that. Other settings, like `enable shadows` say something like:
> Bear in mind that this will force OpenMW to use shaders as if force shaders was enabled.